File: ABC/130/D.py
n, k = map(int, input().split())
a = list(map(int, input().split()))

l, r = 0, 0
curr = 0
res = 0

while l < n:
    while r < n and curr < k:
        curr += a[r]
        r += 1
    if curr >= k:
        res += n - r + 1
    curr -= a[l]
    l += 1

print(res)

File: ABC/130/D.cpp
#include <iostream>
#include <vector>
using namespace std;
using ll = long long;

#define rep(i, n) for (int i = 0; i < (n); i++)

int main() {
  int n;
  ll k;
  cin >> n >> k;
  vector<int> a(n);
  rep(i, n) cin >> a[i];
  int r = 0;
  ll res = 0;
  ll curr = 0;
  rep(l, n) {
    while (r < n && curr + a[r] < k) {
      curr += a[r];
      r++;
    }
    res += r - l;
    curr -= a[l];
  }
  res = (ll)(n+1) * n / 2 - res;
  cout << res << endl;
}
